src/cts/src/TritonCTS.cpp

```diff
@@ -265,12 +265,41 @@ void TritonCTS::countSinksPostDbWrite(
       int receiverX, receiverY;
       iterm->getAvgXY(&receiverX, &receiverY);
       unsigned dist = abs(driverX - receiverX) + abs(driverY - receiverY);
-      bool terminate
-          = fullTree
-                ? (sinks.find(iterm) != sinks.end())
-                : !builder->isAnyTreeBuffer(getClockFromInst(iterm->getInst()));
+      odb::dbInst* inst = iterm->getInst();
+      bool terminate = fullTree
+                           ? (sinks.find(iterm) != sinks.end())
+                           : !builder->isAnyTreeBuffer(getClockFromInst(inst));
+      odb::dbITerm* outputPin = iterm->getInst()->getFirstOutput();
+      bool trueSink = true;
+      if (outputPin && outputPin->getNet() == net) {
+        // Skip feedback loop.  When input pin and output pin are
+        // connected to the same net this can lead to infinite recursion. For
+        // example, some designs have Q pin connected to SI pin.
```

Comment:
Why would we be tracing clock sinks through the Q pin?

Reply:
Actually, it goes from clock pin to Q pin and then to SI pin.   Sink identification becomes a little tricky with multiple clock trees.

Reply:
Doesn't this mean we will miscount the sinks?  What is the trickiness?